examples/hello/lib/hello/app.rb

```diff
@@ -0,0 +1,66 @@
+require "hello/actions"
+
+module Hello
+  class App
+    class Router
+      class NotFound < StandardError; end
+
+      attr_reader :routes
+
+      def initialize(&block)
+        @routes = {}
+        instance_eval(&block)
+      end
+
+      def resource(name, at:, &block)
+        @this_path = at
+        routes[@this_path] = {}
+        instance_eval(&block)
+      end
+
+      def on(req_method, call:)
+        routes[@this_path][req_method] = call
+      end
+
+      def action_for(path, req_method)
+        routes.fetch(path).fetch(req_method)
+      rescue KeyError
+        raise NotFound.new("Can't find route for #{req_method} #{path}")
+      end
+    end
+
+    def call(env)
+      router = Router.new do
```

Comment:
Eventually I would like to declare a router object outside of the `App` object and pass it in. But I would still like the route definitions to have a standard location so we know where to look. I was thinking `my_app/router` would hold a `Router` that inherits from `Wren::Router` and declares the routes in it's init method with a call to `super`. Then you'd pass `MyApp::Router.new` to `MyApp::App.new`.

I want the router to be an object you can instantiate, pass around, add methods to, etc. I also want it to have little or no dependencies so it's quick and cheap to instantiate.

I don't want it to be global either. If something needs routing, an instance of the router should be passed in.